charts/consul/test/unit/mesh-gateway-deployment.bats

```diff
@@ -1783,4 +1783,21 @@ key2: value2' \
   local actual=$(echo "$cmd" |
     yq 'any(contains("-log-level=warn"))' | tee /dev/stderr)
   [ "${actual}" = "true" ]
+}
+
+#--------------------------------------------------------------------
+# security context
+
+@test "meshGateway/Deployment: don't drop ALL capabilities when hostNetwork=true" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -s templates/mesh-gateway-deployment.yaml  \
+      --set 'meshGateway.enabled=true' \
+      --set 'connectInject.enabled=true' \
+      --set 'meshGateway.hostNetwork=true' \
+      . | tee /dev/stderr |
+      yq -r '.spec.template.spec.containers[0].securityContext' | tee /dev/stderr)
+
+  [ $(echo "${actual}" | yq -r '.capabilities.add[0]') = "NET_BIND_SERVICE" ]
```

Comment:
this doesn't test that we don't set `drop: all`, just that we set `add`

Reply:
should be two tests: one that tests we set drop: all by default, another that tests we don't set it when hostnetwork=true